libs/ui-lib/lib/ocm/components/clusterConfiguration/ExternalPlatformDropdown.tsx

```diff
@@ -0,0 +1,142 @@
+import React from 'react';
+import { Dropdown, DropdownItem, DropdownToggle, FormGroup } from '@patternfly/react-core';
+import { CaretDownIcon } from '@patternfly/react-icons';
+import { useField } from 'formik';
+import {
+  NUTANIX_CONFIG_LINK,
+  PlatformType,
+  VSPHERE_CONFIG_LINK,
+  getFieldId,
+} from '../../../common';
+
+const INPUT_NAME = 'platform';
+const fieldId = getFieldId(INPUT_NAME, 'input');
+export type ExternalPlatformType = Extract<PlatformType, 'none' | 'nutanix' | 'oci' | 'vsphere'>;
+
+type ExternalPlatformDropdownProps = {
+  showOciOption: boolean;
+  selectedPlatform: ExternalPlatformType;
+  disabledOciTooltipContent: React.ReactNode;
+  isOciDisabled: boolean;
+  onChange: (selectedPlatform: ExternalPlatformType) => void;
+  dropdownIsDisabled: boolean;
+};
+
+export type ExternalPlatformInfo = {
+  label: string;
+  href: string;
+  tooltip: string;
+};
+
+export const externalPlatformTypes: Record<ExternalPlatformType, ExternalPlatformInfo> = {
+  none: {
+    label: 'None',
+    href: '',
+    tooltip: '',
+  },
+  nutanix: {
+    label: 'Nutanix',
+    href: NUTANIX_CONFIG_LINK,
+    tooltip: '',
+  },
+  oci: {
+    label: 'Oracle  (Requires a custom manifest)',
```

Comment:
For OCI, It is stated that CM are mandatory.
However, both before creating the cluster, or after it's in Draft, I'm able to uncheck the "custom manifests" checkbox which would make OCI not work and could delete the existing manifests.

Reply:
Yes. It's the actual behavior with the external partner integration checkbox. We need to think if the user selects OCI, then when trying to uncheck the custom manifests checkbox maybe an alert could be shown. @nirfarkas WDYT?

Reply:
If it's going to break the installation, then it should not be allowed.
IMO, the custom manifests checkbox should be disabled and selected when OCI is selected. 

Specially, since when you edit the cluster, unchecking the checkbox immediate performs a DELETE on the manifests.

Reply:
@gamli75 we need to understand this point.
If the user selects OCI option, the custom manifest is mandatory? It's going to break the installation if users don't add manifests with OCI option? Thanks.

Reply:
After some discussions, we agree with @nirfarkas  that “Custom manifests” checkbox should be selected and disabled if “Oracle” option was selected.

Reply:
https://redhat-internal.slack.com/archives/CUPJTHQ5P/p1689581850055749